core/trino-main/src/test/java/io/trino/operator/scalar/TestStringFunctions.java

```diff
@@ -241,6 +241,23 @@ public void testReplace()
         assertFunction("REPLACE('::\uD801\uDC2D::', ':', '')", createVarcharType(5), "\uD801\uDC2D"); //\uD801\uDC2D is one character
         assertFunction("REPLACE('\u00D6sterreich', '\u00D6', 'Oe')", createVarcharType(32), "Oesterreich");
 
+        // Test replace (char(x), varchar(y), varchar(z)) and replace (char(x), varchar(y))
+        assertFunction("REPLACE(CHAR 'aaa', 'a', 'aa')", createVarcharType(11), "aaaaaa");
```

Comment:
I'd expect `REPLACE()` to return a data type matching the input, so it's a bit confusing to work with `CHAR`. What are the issues with casting `CHAR` to `VARCHAR(n)`?

```sql
select replace(cast(char 'aaa' as varchar(3)), 'a', 'aa');
```

Reply:
> I'd expect `REPLACE()` to return a data type matching the input, so it's a bit confusing to work with `CHAR`. What are the issues with casting `CHAR` to `VARCHAR(n)`?
> 
> ```sql
> select replace(cast(char 'aaa' as varchar(3)), 'a', 'aa');
> ```

We have tables already defined as char as the column type, and without a coercion, the schema has to be updated to keep using this function. This coercion was there a while ago, related issue #9031